src/SFA.DAS.Payments.ServiceFabric.Core/ServiceBusBatchCommunicationListener.cs

```diff
@@ -0,0 +1,315 @@
+                        foreach (var message in messages)
+                        {
+                            cancellationToken.ThrowIfCancellationRequested();
+                            try
+                            {
+                                var applicationMessage = DeserializeMessage(message);
+                                var key = applicationMessage.GetType();
+                                var applicationMessages = groupedMessages.ContainsKey(key)
+                                    ? groupedMessages[key]
+                                    : groupedMessages[key] = new List<(string, int, object)>();
+                                applicationMessages.Add((message.SystemProperties.LockToken, message.SystemProperties.DeliveryCount, applicationMessage));
+
+                            }
+                            catch (Exception e)
+                            {
+                                logger.LogError($"Error deserialising the message. Error: {e.Message}", e);
+                                //TODO: should use the error queue instead of dead letter queue
+                                await messageReceiver.DeadLetterAsync(message.SystemProperties.LockToken)
+                                    .ConfigureAwait(false);
+                            }
+                        }
+
+                        //var stopwatch = Stopwatch.StartNew();
+                        await Task.WhenAll(groupedMessages.Select(group => ProcessMessages(group.Key, group.Value, messageReceiver, cancellationToken)));
+                        //stopwatch.Stop();
+                        //RecordAllBatchProcessTelemetry(stopwatch.ElapsedMilliseconds, messages.Count);
+                        //pipeLineStopwatch.Stop();
+                        //RecordPipelineTelemetry(pipeLineStopwatch.ElapsedMilliseconds, messages.Count);
+
+                    }
+                    catch (TaskCanceledException)
+                    {
+                        logger.LogWarning("Cancelling communication listener.");
+                        return;
+                    }
+                    catch (OperationCanceledException)
+                    {
+                        logger.LogWarning("Cancelling communication listener.");
+                        return;
+                    }
+                    catch (Exception ex)
+                    {
+                        logger.LogError($"Error listening for message.  Error: {ex.Message}", ex);
+                    }
+                }
+            }
+            finally
+            {
+                if (!messageReceiver.IsClosedOrClosing)
+                    await messageReceiver.CloseAsync();
+                if (!connection.IsClosedOrClosing)
+                    await connection.CloseAsync();
+            }
+        }
+
+        //private void RecordBatchProcessTelemetry(long elapsedMilliseconds, int count)
```

Comment:
Can this commented out code be removed or is it awaiting other components to be migrated before being enabled?

Reply:
I think these are legacy bits of code. They've been removed now.

We have the git history if needed.